### PR TITLE
Make Handler receive request

### PR DESCRIPTION
**Pull request by younger15**

### CODE REVIEW

File: src/Header/RequestHandler.h [hunk 1]
```diff
@@ -19,12 +19,19 @@
...
+    void RequestHandler::WaitForMessage();
+    //Get client file descriptor
+    int GetClient();
     //content of request
     std::unique_ptr<OperateData> data;
     //storing all feature api
     static const std::unordered_map<std::string, APIHandler&> apis;
+private:
+    //file descriptor
+    int client_fd;
```
Comment (Jazz86) [inline src/Header/RequestHandler.h]: please use clang-format. the indent should be 2 space.

File: src/Header/RequestHandler.h [hunk 2]
```diff
@@ -19,12 +19,19 @@
 class RequestHandler
 {
 public:
-    //instantiate instance with request, will parse to OperateData and check with Authenticator and call correspond api.
-    RequestHandler(const std::string &requestData);
+    //instantiate instance with accepted file descriptor, will receive request string and parse to OperateData, then check with Authenticator and call correspond api.
+    RequestHandler(const int &file_descriptor);
```
Comment (Jazz86) [inline src/Header/RequestHandler.h]: why use file descriptor?

Comment (younger15) [reply src/Header/RequestHandler.h]: As mentor said that handler should keep waiting for same user request, or else the MainManager will stock by the user who is connecting but not sending request with multithread.

**Review by Jazz86 — APPROVED** [review]
approve.

File: src/Header/RequestHandler.h (hunk 2 above)
Comment (Jazz86) [reply src/Header/RequestHandler.h]: understood.

Comment (john0312) [reply src/Header/RequestHandler.h]: This is reasonable.

File: src/Header/RequestHandler.h [hunk 3]
```diff
@@ -19,12 +19,19 @@
...
+    //receive request string and send to api
+    void RequestHandler::WaitForMessage();
```
Comment (john0312) [inline src/Header/RequestHandler.h]: You don't need to add `RequestHandler::` in the definition.

File: src/Header/RequestHandler.h [hunk 4]
```diff
@@ -19,12 +19,19 @@
...
+    //Get client file descriptor
+    int GetClient();
```
Comment (john0312) [inline src/Header/RequestHandler.h]: (Optional) `GetClientFd` to be clearer.

File: src/Header/RequestHandler.h [hunk 5]
```diff
@@ -19,12 +19,19 @@
...
     //content of request
     std::unique_ptr<OperateData> data;
     //storing all feature api
     static const std::unordered_map<std::string, APIHandler&> apis;
+private:
+    //file descriptor
```
Comment (john0312) [inline src/Header/RequestHandler.h]: (Optional) `File description for this request` to be clearer.

**Review by john0312 — APPROVED** [review]
Approve assuming comments are addressed.

File: src/Header/RequestHandler.h (hunk 5 above)
Comment (younger15) [reply src/Header/RequestHandler.h]: Done

File: src/Header/RequestHandler.h (hunk 4 above)
Comment (younger15) [reply src/Header/RequestHandler.h]: Done

File: src/Header/RequestHandler.h (hunk 1 above)
Comment (younger15) [reply src/Header/RequestHandler.h]: Done

File: src/Header/RequestHandler.h (hunk 3 above)
Comment (younger15) [reply src/Header/RequestHandler.h]: Done

**Pull request merged**